### PR TITLE
Remove DisplayMode

### PR DESCRIPTION
**Pull request by bugadani**
This PR removes the DisplayMode struct. As far as I can tell, DisplayMode is only used currently as the return value of Builder::connect() and it is immediately transformed into a specific DisplayModeTrait implementation.

This PR also removes the now obsolete RawMode which was closely connected to the above mechanism.

This is only a proposal and I fully expect it to be rejected. Also, the Into trait is not implemented for the display modes, as for some reason I get trait impl conflicts when I attempt to do so.

### CODE REVIEW

**Comment by jamwaffles** [comment]
If this is a superset of #118, please close #118 so we don't get duplicate PRs.

**Comment by bugadani** [comment]
Not so much a superset, but I branched off of that PR in case this one is rejected or that one is accidentally merged :) It's not about duplicate functionality, but incremental changes.

I have one more addition to make: brightness adjustment. Should I include it in this PR as well?

**Comment by jamwaffles** [comment]
> Not so much a superset, but I branched off of that PR

Ok, understood! I notice both commits from the other PR are included in this one though, so it might be a good idea to rebase this PR from master to keep all these PRs more tightly scoped.

> I have one more addition to make: brightness adjustment.

More additions always welcome! I would prefer this to be in a separate PR if you don't mind. For one thing, it helps keep the changelog organised :)

**Comment by bugadani** [comment]
> so it might be a good idea to rebase this PR from master to keep all these PRs more tightly scoped.

Unfortunately, rebasing this to master would cause some conflict so it would be more work than merging #118 and then continuing work on this one. I'll tidy that one up first.

**Comment by jamwaffles** [comment]
Ok, no worries :)

**Comment by therealprof** [comment]
> Unfortunately, rebasing this to master would cause some conflict so it would be more work than merging #118 and then continuing work on this one. I'll tidy that one up first.

That is superweird and unexpected. But it's a good idea to keep seperate things separate anyway.

**Comment by bugadani** [comment]
I've spent some time cleaning up the links. Unfortunately, it looks like the link checker doesn't know about inter-crate link format and fails, however I believe it is the preferred linking format since `cargo doc` automatically checks them when generating the docs.

File: src/lib.rs [hunk 1]
```diff
@@ -2,8 +2,8 @@
 //!
 //! The driver must be initialised by passing an I2C or SPI interface peripheral to the [`Builder`],
 //! which will in turn create a driver instance in a particular mode. By default, the builder
-//! returns a [`RawMode`] instance which isn't very useful by itself. You can coerce the driver
-//! into a more useful mode by calling `into()` and defining the type you want to coerce to. For
+//! returns a [`DisplayProperties`] instance which is a low level interface to manipulate the display properties (e.g. rotation).
+//! You can coerce the driver into a more useful mode by calling `into()` and defining the type you want to coerce to. For
```
Comment (jamwaffles) [inline src/lib.rs:6]: Please wrap this to 100 columns.

While we're here, I'd also suggest rewording the `You can coerce the driver` part to `The driver can be coerced`.

File: src/builder.rs [hunk 2]
```diff
@@ -53,14 +49,16 @@
 //! let interface = display_interface_spi::SPIInterfaceNoCS::new(spi, dc);
 //! let display: TerminalMode<_> = Builder::new().connect(interface).into();
 //! ```
+//!
+//! [`I2CDIBuilder`]: crate::builder::I2CDIBuilder
+//! [`DisplayProperties`]: crate::properties::DisplayProperties
+//! [`GraphicsMode`]: crate::mode::graphics::GraphicsMode
+//! [`TerminalMode`]: crate::mode::terminal::TerminalMode
```
Comment (jamwaffles) [inline src/builder.rs:56]: Much cleaner, thanks. As mentioned in the review comments, please change these links (and all others in the PR) back to relative paths. Linkchecker and manual docs surfing should help here.

**Comment by bugadani** [comment]
I'll make the changes once I'm at a computer, but are the intra-crate links really nightly? I'll also double check that but I think they are stable, at least docs.rs knows about them IIRC.

**Comment by jamwaffles** [comment]
Unfortunately yes, at least until rust-lang/rust#43466 is closed I guess. As for docs.rs, <https://docs.rs/about> states that:

> Docs.rs automatically builds crates' documentation released on crates.io using the nightly release of the Rust compiler.

Which is why they work there.

**Comment by therealprof** [comment]
Nice cleanup. Would be great to land this.

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 ### Changed
 
+- [#119](https://github.com/jamwaffles/ssd1306/pull/119) Remove `DisplayMode` and `RawDisplayMode`
```
Comment (jamwaffles) [inline CHANGELOG.md:11]:
```suggestion
-**(breaking)** [#119](https://github.com/jamwaffles/ssd1306/pull/119) Remove `DisplayMode` and `RawMode`
```

**Comment by bugadani** [comment]
I hope you don't mind if I took the lazy way of pressing the Commit suggestion button 😅

**Comment by jamwaffles** [comment]
No problem - I squash merged this PR anyway :). Thanks for the changes!
